network/struct.go

```diff
@@ -107,6 +109,7 @@ func NewServerIdentity(public abstract.Point, address Address) *ServerIdentity {
 		Public:  public,
 		Address: address,
 		ID:      ServerIdentityID(uuid.NewV5(uuid.NamespaceURL, url)),
+		Cert:    []byte{},
```

Comment:
Mhhhhh. Instead of putting a default value here, and after calling `si.Cert = tls.Cert`, it could be better to have a `NewTLSServerIdentity` taking the cert directly and setting the Cert to nil in this case (or a flag saying this SI does not have any cert in it).